libminifi/include/FlowFileRecord.h

```diff
@@ -32,198 +32,138 @@
-	uint64_t _penaltyExpirationMs;
-	//! Attributes key/values pairs for the flow record
-	std::map<std::string, std::string> _attributes;
-	//! Pointer to the associated content resource claim
-	ResourceClaim *_claim;
-	//! UUID string
-	std::string _uuidStr;
-	//! UUID string for all parents
-	std::set<std::string> _lineageIdentifiers;
-	//! duplicate the original flow file
-	void duplicate(FlowFileRecord *original);
-
-private:
-
-	//! Local flow sequence ID
-	static std::atomic<uint64_t> _localFlowSeqNumber;
-	//! Mark for deletion
-	bool _markedDelete;
-	//! Connection queue that this flow file will be transfer or current in
-	Connection *_connection;
-	//! Orginal connection queue that this flow file was dequeued from
-	Connection *_orginalConnection;
-	//! Logger
-	std::shared_ptr<Logger> logger_;
-	//! Snapshot flow record for session rollback
-	bool _snapshot;
-	// Prevent default copy constructor and assignment operation
-	// Only support pass by reference or pointer
-	FlowFileRecord(const FlowFileRecord &parent);
-	FlowFileRecord &operator=(const FlowFileRecord &parent);
+class FlowFileRecord : public core::Record {
+ public:
+  //! Constructor
+  /*!
+   * Create a new flow record
+   */
+  explicit FlowFileRecord(std::map<std::string, std::string> attributes,
+                          std::shared_ptr<ResourceClaim> claim = nullptr);
+  //! Destructor
+  virtual ~FlowFileRecord();
+  //! addAttribute key is enum
+  bool addKeyedAttribute(FlowAttribute key, std::string value);
+  //! removeAttribute key is enum
+  bool removeKeyedAttribute(FlowAttribute key);
+  //! updateAttribute key is enum
+  bool updateKeyedAttribute(FlowAttribute key, std::string value);
+  //! getAttribute key is enum
+  bool getKeyedAttribute(FlowAttribute key, std::string &value);
+ 
+
+  void setSnapShot(bool snapshot) {
+    snapshot_ = snapshot;
+  }
+
+  FlowFileRecord &operator=(const FlowFileRecord &);
+
+  FlowFileRecord(const FlowFileRecord &parent) = delete;
+
+ protected:
+  /*
+   //! Date at which the flow file entered the flow
+   uint64_t _entryDate;
+   //! Date at which the origin of this flow file entered the flow
+   uint64_t _lineageStartDate;
+   //! Date at which the flow file was queued
+   uint64_t _lastQueueDate;
+   //! Size in bytes of the data corresponding to this flow file
+   uint64_t _size;
+   //! A global unique identifier
+   uuid_t _uuid;
+   //! A local unique identifier
+   uint64_t _id;
+   //! Offset to the content
+   uint64_t _offset;
+   //! Penalty expiration
+   uint64_t _penaltyExpirationMs;
+   //! Attributes key/values pairs for the flow record
+   std::map<std::string, std::string> _attributes;
+   //! Pointer to the associated content resource claim
+   ResourceClaim *_claim;
+   //! UUID string
+   std::string _uuidStr;
+   //! UUID string for all parents
+   std::set<std::string> _lineageIdentifiers;
+   */
+
+ private:
+
+  //! Local flow sequence ID
+  static std::atomic<uint64_t> _localFlowSeqNumber;
```

Comment:
Can you move "_" to end of class data member to adhere to Google C++ Style guide?